example/secure_client.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+
+import asyncio
+import os.path
```

Comment:
totally unrelated to the PR, but generally speaking since you require python-3.4 I suggest you use `pathlib` instead of old-school `os.path`

Reply:
I'm old-school and haven't adopted pathlib yet — mostly because casting `Path` objects back to `str` for APIs that don't accept a `Path` is tedious. 

In this case, it works, so I'm making the change.

Reply:
While we're talking of fancy new features, I'm going to use f-strings as well.

Reply:
well f-strings comes with 3.6 only, so this sounds like too much in the case of your library since, IIUC, it runs on 3.4 and up

Reply:
That's fine - many examples use features that aren't available in Python 3.4; I'm providing separate examples for older Python versions.